Sources/OperatingSystem.swift

```diff
@@ -26,4 +26,210 @@
+    case tooManyLinks(description: String, data: Data)
+    case messageTooLarge(description: String, data: Data)
+    case filenameTooLong(description: String, data: Data)
+    case networkIsDown(description: String, data: Data)
+    case connectionAbortedByNetwork(description: String, data: Data)
+    case networkUnreachable(description: String, data: Data)
+    case tooManyFilesOpenInSystem(description: String, data: Data)
+    case noBufferSpaceAvailable(description: String, data: Data)
+    case noSuchDevice(description: String, data: Data)
+    case noSuchFileOrDirectory(description: String, data: Data)
+    case executableFileFormatError(description: String, data: Data)
+    case noLocksAvailable(description: String, data: Data)
+    case notEnoughSpace(description: String, data: Data)
+    case noMessageOfTheDesiredType(description: String, data: Data)
+    case protocolNotAvailable(description: String, data: Data)
+    case noSpaceLeftOnDevice(description: String, data: Data)
+    case noMessageIsAvailableOnTheStreamHeadReadQueue(description: String, data: Data)
+    case noStreamResources(description: String, data: Data)
+    case notAStream(description: String, data: Data)
+    case functionNotSupported(description: String, data: Data)
+    case socketIsNotConnected(description: String, data: Data)
+    case notADirectory(description: String, data: Data)
+    case directoryNotEmpty(description: String, data: Data)
+    case notASocket(description: String, data: Data)
+    case notSupported(description: String, data: Data)
+    case inappropriateInputOutputControlOperation(description: String, data: Data)
+    case noSuchDeviceOrAddress(description: String, data: Data)
+    case operationNotSupportedOnSocket(description: String, data: Data)
+    case valueTooLargeToBeStoredInDataType(description: String, data: Data)
+    case operationNotPermitted(description: String, data: Data)
+    case brokenPipe(description: String, data: Data)
+    case protocolError(description: String, data: Data)
+    case protocolNotSupported(description: String, data: Data)
+    case protocolWrongTypeForSocket(description: String, data: Data)
+    case resultTooLarge(description: String, data: Data)
+    case readOnlyFileSystem(description: String, data: Data)
+    case invalidSeek(description: String, data: Data)
+    case noSuchProcess(description: String, data: Data)
+    case streamTimeout(description: String, data: Data)
+    case connectionTimedOut(description: String, data: Data)
+    case textFileBusy(description: String, data: Data)
+    case operationWouldBlock(description: String, data: Data)
+    case crossDeviceLink(description: String, data: Data)
+    case other(description: String)
+
+}
+
+extension SystemError {
+    
+    public static func lastErrorDescription() -> String {
+        return String(validatingUTF8: strerror(errno))!
+    }
+    
+    public static func lastError(with data: Data) -> SystemError {
```

Comment:
Besides than being able to do:
```swift
enum SystemError {
  case connectionTimedOut: ETIMEDOUT
}
```

Also, `RawRepresentable` doesn't allow associated types in the enum cases.